once/__init__.py

```diff
@@ -182,26 +233,24 @@ def _inspect_function(self, func: collections.abc.Callable):
     # bound version of the function to the object or class.
     def __get__(self, obj, cls):
         if self.is_classmethod:
-            func = functools.partial(self.func, cls)
-            return functools.partial(self._execute_call_once, func)
+            return self._callable(functools.partial(self.func, cls))
         if self.is_staticmethod:
-            return functools.partial(self._execute_call_once, self.func)
-        return functools.partial(self._execute_call_once, self.func, obj)
+            return self._callable(self.func)
+        return self._callable(functools.partial(self.func, obj))
```

Comment:
Do I understand this right - if this is a wrapper on a class method, then the first argument "self" should be the target class, which in this case is obj? Obj (the word, not the functionality) seems to be an instance of a class instead of the actual class to me!

Reply:
Python has concepts called `classmethod` and `staticmethod` - if it is classmethod, it is the class itself that it operates on, as opposed to an instantiated object of that class! For a bit more reading: https://medium.com/codex/python-class-methods-class-vs-instance-vs-static-methods-96d075d27c68